src/transformers/modeling_bert.py

```diff
@@ -760,14 +775,53 @@ def forward(
             encoder_attention_mask=encoder_extended_attention_mask,
             output_attentions=output_attentions,
             output_hidden_states=output_hidden_states,
+            return_tuple=return_tuple,
         )
         sequence_output = encoder_outputs[0]
         pooled_output = self.pooler(sequence_output)
 
-        outputs = (sequence_output, pooled_output,) + encoder_outputs[
-            1:
-        ]  # add hidden_states and attentions if they are here
-        return outputs  # sequence_output, pooled_output, (hidden_states), (attentions)
+        if return_tuple:
+            return (sequence_output, pooled_output) + encoder_outputs[2:]
+
+        return EncoderOutputWithPooling(
+            last_hidden_state=sequence_output,
+            pooler_output=pooled_output,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
```

Comment:
If you feel like it (and it's possible) think it could be interesting to maybe extend the approach to the internal tuples inside the models which are also a source of bugs sometimes (was the case for T5 for instance).

Reply:
Ok. Those can probably be regular `namedtuple` (like in the reformer model file) or I can use the same kind of `ModelOuput` if we want consistency.

Reply:
agree I think we can use regular `namedtuples` inside the models as is already done in Reformer e.g.